src/data/auroradb.ts

```diff
@@ -691,3 +692,39 @@ export const getPassengerTypeByNameAndNocCode = async (
         throw new Error(`Could not retrieve passenger type by nocCode from AuroraDB: ${error}`);
     }
 };
+
+type SavedPassengerType = { group: GroupPassengerType[]; single: PassengerType[] };
+export const getPassengerTypesByNocCode = async <T extends keyof SavedPassengerType>(
+    nocCode: string,
+    type: T,
+): Promise<SavedPassengerType[T]> => {
```

Comment:
Do we need to use generics here or can we have the 2 types of data with a | between them?

Reply:
It needs to be generic so the compiler know what return type to expect based on the input. i.e if you ask for a "group" you will get a group array back